app/Model/CoMessageTemplate.php

```diff
@@ -295,25 +300,31 @@ public function templateSend($id, $toEmail, $substituteWith = array()) {
-      'NEW_AFFIL'        => $substituteWith['NEW_AFFIL'] ?? "faculty",
-      'CO_PERSON'        => $substituteWith['CO_PERSON'] ?? "John Doe",
-      'ORIG_COU'         => $substituteWith['ORIG_COU'] ?? "",
-      'NEW_COU'          => $substituteWith['NEW_COU'] ?? "",
-      'ORIG_STATUS'      => $substituteWith['ORIG_STATUS'] ?? "Active",
-      'NEW_STATUS'       => $substituteWith['NEW_STATUS'] ?? "Grace Period",
-      'DAYS_TO_EXPIRY'   => $substituteWith['DAYS_TO_EXPIRY'] ?? "5",
-      'POLICY_DESC'      => $substituteWith['POLICY_DESC'] ?? "Expiration Policy Description",
-      'SPONSOR'          => $substituteWith['SPONSOR'] ?? "Bob Builder",
-      'TITLE'            => $substituteWith['TITLE'] ?? "Mr",
-      'VALID_FROM'       => $substituteWith['VALID_FROM'] ?? "YYYY/MM/DD",
-      'VALID_THROUGH'    => $substituteWith['VALID_THROUGH'] ?? "YYYY/MM/DD",
-      'TOKEN'            => $substituteWith['TOKEN'] ?? "123456"
+      'CO_NAME'              => $co_name,
+      'ACTOR_NAME'           => $substituteWith['ACTOR_NAME'] ?? 'template_test',
+      'NOTIFICATION_URL'     => $substituteWith['NOTIFICATION_URL'] ?? '#',
+      'SOURCE_URL'           => $substituteWith['SOURCE_URL'] ?? '#',
+      'INVITE_URL'           => $substituteWith['INVITE_URL'] ?? '#',
+      'ORIG_AFFIL'           => $substituteWith['ORIG_AFFIL'] ?? "member",
+      'NEW_AFFIL'            => $substituteWith['NEW_AFFIL'] ?? "faculty",
+      'CO_PERSON'            => $substituteWith['CO_PERSON'] ?? "John Doe",
+      'ORIG_COU'             => $substituteWith['ORIG_COU'] ?? "",
+      'NEW_COU'              => $substituteWith['NEW_COU'] ?? "",
+      'ORIG_STATUS'          => $substituteWith['ORIG_STATUS'] ?? "Active",
+      'NEW_STATUS'           => $substituteWith['NEW_STATUS'] ?? "Grace Period",
+      'DAYS_TO_EXPIRY'       => $substituteWith['DAYS_TO_EXPIRY'] ?? "5",
+      'POLICY_DESC'          => $substituteWith['POLICY_DESC'] ?? "Expiration Policy Description",
+      'SPONSOR'              => $substituteWith['SPONSOR'] ?? "Bob Builder",
+      'TITLE'                => $substituteWith['TITLE'] ?? "Mr",
+      'VALID_FROM'           => $substituteWith['VALID_FROM'] ?? "YYYY/MM/DD",
+      'VALID_THROUGH'        => $substituteWith['VALID_THROUGH'] ?? "YYYY/MM/DD",
+      'TOKEN'                => $substituteWith['TOKEN'] ?? "123456",
+      'ENROLLMENT_NAME'      => $substituteWith['ENROLLMENT_NAME'] ?? "Join vo.example.org COU",
+      'ORIG_PETITION_STATUS' => $substituteWith['ORIG_PETITION_STATUS'] ?? "Created",
+      'NEW_PETITION_STATUS'  => $substituteWith['NEW_PETITION_STATUS'] ?? "Pending Approval",
+      'ENROLLEE_NAME'        => $substituteWith['ENROLLEE_NAME'] ?? "John Doe",
+      'SOURCE_URL'           => $substituteWith['SOURCE_URL'] ?? "http://example.org/registry/petition/view/1",
+      'NOTIFICATION_URL'     => $substituteWith['NOTIFICATION_URL'] ?? "http://example.org/registry/co_notifications/view/1",
```

Comment:
I'm not sure where these default values came from (I guess I missed the PR that added them) but we shouldn't be using these defaults as they might confuse admins who receive them. If we need a default text, it should just be "(unknown)" or "(?)" to indicate that something went wrong.

Reply:
These default values are used only when testing a template. They do not interfere with the rest of the flow.

Reply:
These default values are used only when testing a template. They do not interfere with the rest of the flow. The default, dummy value is needed because when testing the template there are no meaningful data available.

Reply:
ok. the default values for testing should be defined somewhere closer to the testing code then, but that's not a topic for this PR.